### PR TITLE
Migrating from sphinx to Material for MkDocs

### PR DESCRIPTION
**Pull request by DeaMariaLeon**
This is a work in progress in case you want to see locally what I have done. 
As you know, if you download my branch, one can see the docs locally running `mkdocs serve`.

Please note that these points are still pending before starting the next step of adding more documentation.

-  ~~Cross-reference links.~~

- ~~The rest of the main `rst` pages like install, construct, etc,  are still missing.~~ But need links fixed

- ~~Some format on the API documents need to be fixed as well. Like the area of the examples.~~

- ~~Headers,~~ logo, ~~footers~~ need to be added.

- ~~Website needs to be deployed with mkdocs instead of Sphinx~~

@hameerabbasi @mtsokol

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+- literate-nav:
+    nav_file: SUMMARY.md
+- section-index
+- mkdocstrings:
+    handlers:
+          python:
+            import:
+            - https://numpy.org/doc/stable/objects.inv
+            - https://docs.python.org/3/objects.inv
+            - https://docs.scipy.org/doc/scipy-1.14.0/objects.inv
```
Comment (DeaMariaLeon) [inline mkdocs.yml:35]: Does someone know where the general `objects.inv` for `scipy` is located? (to remove the hardcoded version) 
The links for `python` and `scipy` are not like the ones on `config.py` for Sphinx.

Comment (hameerabbasi) [reply mkdocs.yml:36]: The link is: https://docs.scipy.org/doc/scipy/objects.inv

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -65,10 +65,10 @@ jobs:
           cache: 'pip'
       - name: Install package
         run: |
-          pip install -e '.[docs]'
+          - run: pip install -r docs/requirements.txt -e .
```
Comment (hameerabbasi) [inline .github/workflows/ci.yml:68]: Maybe undo these changes? This makes the YAML file invalid.

Comment (DeaMariaLeon) [reply .github/workflows/ci.yml:68]: Thanks!

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -63,9 +63,7 @@ jobs:
         with:
           python-version: '3.10'
           cache: 'pip'
-      - name: Install package
-        run: |
-          - run: pip install -r docs/requirements.txt -e .
+      - run: pip install -r docs/requirements.txt -e .
```
Comment (hameerabbasi) [inline .github/workflows/ci.yml:66]:
```suggestion
      - run: pip install -e ".[docs]"
```

This is equivalent and simpler. 😄

Comment (DeaMariaLeon) [reply .github/workflows/ci.yml:66]: I was doing it like the documentations on Read the docs, to test first.

File: .readthedocs.yml [hunk 4]
```diff
@@ -1,16 +1,22 @@
...
+  fail_on_warning: false
 
+# Optionally declare the Python requirements required to build your docs
 python:
   install:
   - method: pip
     path: .
-    extra_requirements:
-      - docs
+  - requirements: docs/requirements.txt
```
Comment (hameerabbasi) [inline .readthedocs.yml:22]: This can be rolled back if you went back to the `pyproject.toml`.

File: docs/construct.md [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+You can construct [COO][sparse.COO] arrays from coordinates and value data.
+
+The `cords` parameter contains the indices where the data is nonzero,
+and the `data` parameter contains the data corresponding to those indices.
+For example, the following code will generate a $5 \times 5$ diagonal
+matrix:
+
+```python
+
+   >>> import sparse
```
Comment (hameerabbasi) [inline docs/construct.md:14]: Is the indentation in these cases necessary? If not, I'd remove it.

File: docs/construct.md [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+```python
+
+   s[1, 2, 1]  # 5
+   s[5, 1, 1]  # 0
+   s[[0, 3], [0, 4], [0, 1]] # <DOK: shape=(2,), dtype=float64, nnz=2, fill_value=0.0>
+```
+
+## Converting [COO][sparse.COO] objects to other Formats
+
+[COO][sparse.COO] arrays can be converted to [Numpy arrays][numpy.ndarray],
```
Comment (hameerabbasi) [inline docs/construct.md:225]: For links to API, I'd prefer the fully qualified name, and perhaps in monospace font (using backticks in Markdown)

File: docs/contributing.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+[mkdocs-gen-files plugin](https://oprypin.github.io/mkdocs-gen-files)
+to generate API references.
+
+To build the documentation, you can run
+
+```bash
+
+   mkdocs build
+   mkdocs serve
+```
```
Comment (hameerabbasi) [inline docs/contributing.md:98]: Thanks for changing this!

File: docs/construct.md (hunk 6 above)
Comment (hameerabbasi) [reply docs/construct.md:225]: You might be able to use regex find/replace to edit most of these.

**Review by hameerabbasi — CHANGES_REQUESTED** [review]
Some minor changes, but LGTM overall. Thanks for all the hard work, @DeaMariaLeon!

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -63,12 +63,10 @@ jobs:
...
           python-version: '3.10'
           cache: 'pip'
-      - name: Install package
-        run: |
-          pip install -e '.[docs]'
+      - run: pip install -e ".[docs]"
       - name: Run tests
         run: |
-          sphinx-build -W -b html docs/ _build/html
+          mkdocs build
```
Comment (hameerabbasi) [inline .github/workflows/ci.yml:69]: This entire job can be removed now.

File: docs/construct.md (hunk 6 above)
Comment (DeaMariaLeon) [reply docs/construct.md:225]: > For links to API, I'd prefer the fully qualified name,

Could you please give me an example of this? I did it to show the links with the same name as the original sphinx website.

Comment (DeaMariaLeon) [reply docs/construct.md:225]: >in monospace font (using backticks in Markdown)

If I use backticks it doesn't resolve the link (it shows [COO][sparse.COO] as opposed to COO).

File: docs/construct.md (hunk 5 above)
Comment (DeaMariaLeon) [reply docs/construct.md:14]: Do you mean on all the code boxes? (all files)?

File: docs/construct.md (hunk 6 above)
Comment (hameerabbasi) [reply docs/construct.md:225]: I guess one of the following should work:

```
[][sparse.COO]
[`sparse.COO`][sparse.COO]
```

File: docs/construct.md (hunk 5 above)
Comment (hameerabbasi) [reply docs/construct.md:14]: Yes.

File: docs/construct.md (hunk 6 above)
Comment (hameerabbasi) [reply docs/construct.md:225]: > Could you please give me an example of this? I did it to show the links with the same name as the original sphinx website.

The full name starting with `sparse`.

Comment (hameerabbasi) [reply docs/construct.md:225]: We agreed to push this out as a bonus task to the end.

**Comment by DeaMariaLeon** [comment]
I think I addressed all the requested changes. Please let me know if something is missing.

File: docs/quickstart.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
+# Getting Started
+
+## Install
+
+If you haven't already, install the ``sparse`` library
```
Comment (hameerabbasi) [inline docs/quickstart.md:5]:
```suggestion
If you haven't already, install the `sparse` library
```

File: docs/quickstart.md [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+x[x < 0.9] = 0  # fill most of the array with zeros
+
+s = sparse.COO(x)  # convert to sparse array
+```
+
+These store the same information and support many of the same operations,
+but the sparse version takes up less space in memory
+
+```python
+
```
Comment (hameerabbasi) [inline docs/quickstart.md:30]:
```suggestion
```

File: docs/quickstart.md [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+For more efficient ways to construct sparse arrays,
+see documentation on [Construct sparse arrays][construct-sparse-arrays].
+
+## Compute
+
+Many of the normal Numpy operations work on [`sparse.COO`][] objects just like on [`numpy.ndarray`][] objects.
+This includes arithmetic, [`numpy.ufunc`][] operations, or functions like tensordot and transpose.
+
+```python
+
```
Comment (hameerabbasi) [inline docs/quickstart.md:48]:
```suggestion
```

File: docs/quickstart.md [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+>>> np.sin(s) + s.T * 1
+<COO: shape=(100, 100, 100), dtype=float64, nnz=189601, fill_value=0.0>
+```
+
+However, operations which map zero elements to nonzero will usually change the fill-value
+instead of raising an error.
+
+```python
+
```
Comment (hameerabbasi) [inline docs/quickstart.md:57]:
```suggestion
```

File: docs/quickstart.md [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+>>> y = s + 5
+<COO: shape=(100, 100, 100), dtype=float64, nnz=100246, fill_value=5.0>
+```
+
+However, if you're sure you want to convert a sparse array to a dense one,
+you can use the ``todense`` method (which will result in a [`numpy.ndarray`][]):
+
+```python
+
```
Comment (hameerabbasi) [inline docs/quickstart.md:66]:
```suggestion
```

**Review by hameerabbasi — CHANGES_REQUESTED** [review]
Some minor missed changes, then we're all good.

**Review by hameerabbasi — APPROVED** [review]
Thanks for this heavy lift and your patience, @DeaMariaLeon! Let's get this in and iterate on it at this point.

**Comment by DeaMariaLeon** [comment]
Thank you @hameerabbasi !
